### ISSUE TITLE
If i have four button and four tab strip child view controller then if i click on seocond button it should open directly second child view controller.How to achieve this.

### ISSUE THREAD

**Issue by nishantapps** [xmartlabs/XLPagerTabStrip#658]
Before submitting issues ...

- Make sure you are using XLPagerTabStrip [latest release](https://github.com/xmartlabs/XLPagerTabStrip/releases) or master branch version.
- Make sure your Xcode version is the latest stable one.
- Check if the issue was [already reported or fixed](https://github.com/xmartlabs/XLPagerTabStrip/issues?utf8=%E2%9C%93&q=is%3Aissue). We add labels to each issue in order to easily find related issues. If you found a match add a brief comment "I have the same problem" or "+1".
- Please do not use the issue tracker for personal support requests. Stack Overflow is a better place for that where a wider community can help you!

When submitting issues, please provide the following information to help maintainers to fix the problem faster:

- Environment: XLPagerTabStrip, Xcode and iOS version you are using.
- In case of reporting errors, provide Xcode console output of stack trace or code compilation error.
- Any other additional detail such as example code that you think it would be useful to understand, reproduce and solve the problem.

**Comment by mobilewebstar** [comment]
I have the same problem, too. how can I fix it?

**Comment by nishantapps** [comment]
i found the solution.

1.first save an index path in user defaults suppose you want go to which child controller like in code where you are pushing to view controller

let check = UserDefaults.standard.set(1, "check") //in this case i want to go to child at path 1 which is child2 

2.second step
In Parent controller, implement view didAppear and check 

   override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)

        if check == 1 {
           moveToViewController(at:1 )
        }
    }

----Hope this help you---

**Comment by mobilewebstar** [comment]
I already tried it. but when clicked the second button, I'm taken to the second child view controller after ended up on the first child view controller though I set the argument "animated" to false.

**Comment by nishantapps** [comment]
Yeah its animation i am also facing the animation problem.It is going but it is animating.

**Comment by mobilewebstar** [comment]
I found best solution.
https://github.com/Sepicat/XLPagerTabStrip/commit/81f3b87c884765444a404b2f22afd1bdd11df8f0#commitcomment-31279765

**Comment by nishantapps** [comment]
so where we have to call that func.

> I found best solution.
> [Sepicat@81f3b87#commitcomment-31279765](https://github.com/Sepicat/XLPagerTabStrip/commit/81f3b87c884765444a404b2f22afd1bdd11df8f0#commitcomment-31279765)

So where exactly we have to call or is working properly.

**Comment by mobilewebstar** [comment]
in both viewDidLoad() and viewWillAppear()

**Comment by nishantapps** [comment]
> in both viewDidLoad() and viewWillAppear()

i have tried but not successful. can you be specific by example.

**Comment by nishantapps** [comment]
> in both viewDidLoad() and viewWillAppear()

I mean where to call in child or parent. I am also getting error of "Cannot assign to property: 'currentIndex' setter is inaccessible".
